slmsuite/hardware/cameras/simulated.py

```diff
@@ -0,0 +1,219 @@
+
+    Note
+    ~~~~
+    For fastest simulation, initialize :class:`SimulatedCam` with a 
+    :class:`~slmsuite.hardware.slms.simulated.SimulatedSLM` *only*. Simulated camera images
+    will directly sample the (quickly) computed SLM far-field (`"knm"`) instead of interpolating 
+    the SLM's far-field intensity at each camera pixel location (i.e. `"knm"`->"ij" basis change),
+    which may also require additional padding (computed automatically upon initialization) for
+    sufficient resolution.  
+
+    Attributes
+    ----------
+    resolution : tuple
+        (width, height) of the SLM in pixels.
+    exposure : float
+        Digital gain value to simulate exposure time. Directly proprtional to imaged power.
+    xgrid : ndarray
+        Pixel column number (``"ij"`` basis) used for far-field interpolation.
+    ygrid : ndarray
+        Pixel row number (``"ij"`` basis) used for far-field interpolation.
+    f_eff : float
+        Effective focal length (in `basis` units) of the
+        optical train separating the Fourier-domain SLM from the camera.
+
+        Important
+        ~~~~~~~~~
+        The normalized unit for `f_eff` is pixels/radian, i.e. the units of :math:`M` matrix 
+        elements required to convert normalized angles/:math:`k`-space coordinates to camera
+        pixels in the ``"ij"`` basis.
+        See :meth:`~slmsuite.hardware.cameraslms.FourierSLM.kxyslm_to_ijcam` for additional 
+        details. To convert to true distance units (e.g., `"um"`), multiply `f_eff` by the the
+        pixel size in the same dimensions.
+        As noted in :meth:`~slmsuite.hardware.cameraslms.get_effective_focal_length`, non-square
+        pixels therefore imply different effective focal lengths along each axis when using
+        true distance units.
+    shape_padded : tuple
+        Size of the FFT computational space required to faitfully reproduce the far-field at
+        full camera resolution.
+
+    """
+
+    def __init__(self, slm, resolution=None, f_eff=None, theta=0, offset=None, basis="ij", **kwargs):
+        """
+        Initialize simulated camera.
+
+        Parameters
+        ----------
+        slm : :class:`~slmsuite.hardware.slms.simulated.SimulatedSLM`
+            Simulated SLM creating the image.
+        resolution : tuple
+            See :attr:`resolution`. If ``None``, defaults to the resolution of `slm`.
+        f_eff : float
+            See :attr:`f_eff`. If `None`, defaults to the minimum focal length for
+            which the camera is fully contained within the SLM's accessible Fourier space.
+        theta : float
+            Rotation angle (in radians, ccw) of the camera from the SLM axis.
+            Defaults to 0 (i.e., aligned with the SLM).
+        offset : tuple
+            Lateral displacement (in `basis` units) of the camera center from `slm`'s 
+            optical axis. If ``None``, defaults to 0 offset.
```

Comment:
It might be worth considering making a helper function for constructing affine transformations with offset, rotation, and scale. I think simulated would be simpler and more general if it used an affine transformation, but user-friendliness can still be maintained with a helper function.

Reply:
@ichristen Are you suggesting to feed in the same (M,b) from "kxy"->"ij" transformations, apply the inverse, and then do the interpolation in "kxy" space? Or to apply a forward affine transformation to map "ij" -> "xy"?

Reply:
Yeah, probably the same {"M", "b"}, if that's the option that is most uniform with the rest of the package.

Reply:
Looking at this more, the f_eff scale term is a bit of a pain for the user to deal with in the way I was thinking. I think your implementation is the right approach, with just handling everything inside.

Reply:
Done! Needs hardware testing and a digital twin method now

Reply:
Completed; see simulation tutorial for sample uses.